src/zenml/environment.py

```diff
@@ -158,8 +171,8 @@ def in_docker() -> bool:
             container, `False` otherwise.
         """
         # TODO [ENG-167]: Make this more reliable and add test.
-        if INSIDE_ZENML_CONTAINER:
-            return True
+        if not INSIDE_ZENML_CONTAINER:
```

Comment:
I think this logic is not optimal now, as something like the following won't be detected by either of `in_container` or `in_docker` anymore.

```bash
docker run -it ubuntu bash
pip install zenml
zenml stack register ...
```

Reply:
Right, this is for the case when the CLI is also run in a container. I'll remove the extra condition.

Reply:
Done